docs/error_codes.yaml

```diff
@@ -0,0 +1,151 @@
+#
+# Copyright (C) 2024, Pelican Project, Morgridge Institute for Research
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you
+# may not use this file except in compliance with the License.  You may
+# obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# This file contains structured documentation about the Pelican parameters.
+# While it is somewhat human-readable, it is meant to help with the documentation
+# generation.
+
+############################
+#    Client Error Codes    #
+############################
+
+############################
+#     Parameter Errors     #
+############################
+---
+type: Parameter
+code: 1000
+clientExitCode: 4
+description: >-
+  If the client failed to start, or was started with invalid parameters, or otherwise believes what it
+  was asked to do is impossible or the request itself is malformed.
+retryable: false
+
+############################
+#    Resolution Errors     #
+############################
+---
+type: Resolution
+code: 2000
+clientExitCode: 5
+description: >-
+  Indicates that the client failed to even attempt to contact the server
+retryable: false
+
+############################
+#      Contact Errors      #
+############################
+---
+type: Contact
+code: 3000
+clientExitCode: 6
+description: >-
+  The client attempted to contact the server at its resolved address and failed to do so
+retryable: false
```

Comment:
In the context of these errors, we should think carefully about how we define "retryable". I'd love to see the definition of that somewhere in this doc.

If we resolved the hostname for these servers and tried to contact them but couldn't, it could mean a blip in the network wherever this job is run and that retrying after a few seconds might work. It could also be that the site has a network issue and that the request is retryable, but only from a different site. These are different definitions of retryable. Do we have a way of distinguishing? If not, how do we decide these constitute a non-retryable error?

Reply:
Good idea! Added a comment explaining retryable. Basically, it is used in the condor sense as it could work if we retry the entire job. The job can either succeed by:
1. Landing at the same site that is now in a better state
2. Landing at a different site